clients/src/main/java/org/apache/kafka/common/network/ChannelState.java

```diff
@@ -43,14 +45,45 @@
  *   <li>PLAINTEXT Good path: NOT_CONNECTED => READY => LOCAL_CLOSE</li>
```

Comment:
nit:  above "States for KafkaChannel" doc is missing description for new states?